### PR TITLE
Fix daily / CI failures by bumping the Test Framework to 1.4.2.Beta9 and migrating package type configuration property

### PR DESCRIPTION
**Pull request by michalvavrik**
### Summary

As a follow up of the https://github.com/quarkusio/quarkus/pull/39295 we should adapt our tests. While fallback config interceptors are in place, we need to sync properties set with how the framework detects them and works with them.

Please select the relevant options.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Dependency update
- [ ] Refactoring
- [ ] Backport
- [ ] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [x] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by michalvavrik** [comment]
@gtroitsk OCP JVM failures are not related because OTel condition seems to fail as test is badly written (I'll provide fix in a separate PR) and Infinispan I can't see related to changes listed in the https://github.com/quarkus-qe/quarkus-test-framework/pull/1069.

**Comment by michalvavrik** [comment]
Ah, there is a difference in the traces after all, but it's not related. Will need to look it up.

**Comment by michalvavrik** [comment]
Found it, caused by https://github.com/quarkusio/quarkus/pull/39534. I'll open new PR.

**Comment by gtroitsk** [comment]
@michalvavrik can you analyse please what caused failures in `OperatorOpenShiftInfinispanObjectsIT`?

**Comment by michalvavrik** [comment]
> @michalvavrik can you analyse please what caused failures in `OperatorOpenShiftInfinispanObjectsIT`?

The question is, how is it relevant to the changes here? There is so many failures I need to fix, this is not on the top of my list. I run `OperatorOpenShiftInfinispanObjectsIT` from my computer and I can reproduce the failure on the current main, which means that it's not related to this PR.

I plan to debug it tomorrow and open upstream issue over class cast exception.

**Comment by gtroitsk** [comment]
> > @michalvavrik can you analyse please what caused failures in `OperatorOpenShiftInfinispanObjectsIT`?
> 
> The question is, how is it relevant to the changes here? There is so many failures I need to fix, this is not on the top of my list. I run `OperatorOpenShiftInfinispanObjectsIT` from my computer and I can reproduce the failure on the current main, which means that it's not related to this PR.
> 
> I plan to debug it tomorrow and open upstream issue over class cast exception.

In that case it is ok, thank you for testing and confirmation. I will merge this PR

**Review by gtroitsk — APPROVED** [review]
Thank you!

File: README.md [hunk 1]
```diff
@@ -151,7 +151,7 @@ Be sure that GraalVM is installed by running the following command, otherwise yo
 User: `Deploy in OpenShift the module http-minimum compiled with my local GraalVM in order to build my application`
 
 ```shell
-mvn clean verify -Popenshift -Dall-modules -Dquarkus.package.type=native -pl http/http-minimum
+mvn clean verify -Popenshift -Dall-modules -Dquarkus.package.jar.type=native -pl http/http-minimum
```
Comment (rsvoboda) [inline README.md:154]: I think this is wrong, `fast-jar,legacy-jar,uber-jar,mutable-jar` should be the allowed only ones @michalvavrik

Comment (michalvavrik) [reply README.md:154]: You are right, luckily it's just a README, I'll create a fix.

**Pull request merged**